src/zmqsubscriber/zmqsubscriber.go

```diff
@@ -155,6 +118,99 @@ func (z *ZMQSubscriber) Run() error {
 	return nil
 }
 
+func (z *ZMQSubscriber) processMessage(topic string, payload [][]byte) error {
+	// TODO: use GetTime() and allow other time sources (eg NTP-corrected)
+	firstSeen := time.Now().UTC()
+
+	switch topic {
+	case topicHashTx:
+		tx, err := parseTransactionHash(firstSeen, payload)
+		if err != nil {
+			return err
+		}
+		z.IncomingTx <- *tx
+	case topicRawTxWithFee:
+		tx, err := parseTransaction(firstSeen, payload)
+		if err != nil {
+			return err
+		}
+		z.IncomingTx <- *tx
+	case topicRawBlock:
+		block, err := parseBlock(firstSeen, payload)
+		if err != nil {
+			return err
+		}
+		z.IncomingBlocks <- *block
+	default:
+		return fmt.Errorf("unknown topic %s", topic)
+	}
+
+	return nil
+}
+
+// Stop sets the cancel flag to true. The ZMQSubscriber is stopped after it
+// finishes receiving a message or reaches the timeout.
 func (z *ZMQSubscriber) Stop() {
 	z.cancel = true
 }
+
+func parseTransactionHash(firstSeen time.Time, payload [][]byte) (*types.Transaction, error) {
+	if len(payload) != 2 {
+		return nil, fmt.Errorf("unexpected payload length: expected len((tx hash, sequence) == 2 but got len(payload) == %d", len(payload))
```

Comment:
`expected len([txhash, sequence]) == 2`

Reply:
Maybe I keep fixing up the wrong commit? Changed this three times now...

Reply:
Oh I see. It's in `parseTransactionHash`. Going to remove that one in an extra commit.